src/exec/mod.rs

```diff
@@ -192,6 +208,33 @@ impl CommandTemplate {
             ExitCode::Success
         }
     }
+
+    /// Helper function to add found paths as arguments to `cmd`.
+    fn add_to_cmd<I>(cmd: &mut Command, arg: &ArgumentTemplate, paths: &mut I)
+        where
+            I: Iterator<Item = String>,
+    {
+        for path in paths {
+            cmd.arg(arg.generate(&path).as_ref());
+        }
+    }
+}
+
+/// This function behaves exactly like the standard library `take` for iterators,
+/// except that the original iterator is not consumed in the process.
```

Comment:
Isn't that what [`Iterator::peekable()`](https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.peekable) is for?

Reply:
As far as I know there is no equivalent of this function in the standard library: could you elaborate on how you wanted to use `peek`?